modules/programs/git-credential-oauth.nix

```diff
@@ -0,0 +1,33 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.git-credential-oauth;
+
+in {
+  meta.maintainers = [ maintainers.tomodachi94 ];
+
+  options = {
+    programs.git-credential-oauth = {
+      enable = mkEnableOption "Git authentication handler for OAuth";
+
+      package = mkPackageOption pkgs "git-credential-oauth" { };
+
+      enableGitCredentialHelper =
+        mkEnableOption "the Git credential handler for OAuth" // {
+          default = true;
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    programs.git.extraConfig.credential.helper =
+      mkIf cfg.enableGitCredentialHelper
+      [ "${cfg.package}/bin/git-credential-oauth" ];
```

Comment:
Is there any situation where you would _not_ want to do this? That is, maybe `enableGitCredentialHelper` could be removed?

Reply:
Actually, good point; the only real reason for enabling this module is for the credential helper. (If they want the package in their `$PATH`, they can just add it to `home.packages`. I'll update this when I have a few minutes :)

Reply:
Added in latest push.